pandas/core/missing.py

```diff
@@ -21,11 +21,16 @@ def mask_missing(arr, values_to_mask):
     Return a masking array of same size/shape as arr
     with entries equaling any member of values_to_mask set to True
     """
-    if not isinstance(values_to_mask, (list, np.ndarray)):
+    if isinstance(values_to_mask, np.ndarray):
+        mask_type = values_to_mask.dtype.type
+    elif isinstance(values_to_mask, list):
```

Comment:
what did this break?

yes, testing the first value is wrong (as it could also be 0-len), further it might have mixed values anyhow.

show me a test that broke?